src/Microsoft.Health.Fhir.Api/Extensions/HttpRequestExtensions.cs

```diff
@@ -24,5 +27,36 @@ public static bool IsFhirRequest(this HttpRequest request)
             return !request.Path.StartsWithSegments(KnownRoutes.HealthCheck, StringComparison.InvariantCultureIgnoreCase) &&
                    !request.Path.StartsWithSegments(KnownRoutes.CustomError, StringComparison.InvariantCultureIgnoreCase);
         }
+
+        /// <summary>
+        /// Return a group of search parameters from a query string.
+        /// </summary>
+        /// <param name="request">The request.</param>
+        public static IReadOnlyList<Tuple<string, string>> GetQueriesForSearch(this HttpRequest request)
+        {
+            EnsureArg.IsNotNull(request, nameof(request));
+
+            if (request.QueryString.HasValue)
+            {
+                NameValueCollection queryStringParameters =
+                    HttpUtility.ParseQueryString(
+                            request.QueryString.Value);
+
+                List<Tuple<string, string>> queries = new List<Tuple<string, string>>();
+                foreach (string key in queryStringParameters.AllKeys)
+                {
+                    foreach (var queryStringValue in queryStringParameters.GetValues(key))
+                    {
+                        queries.Add(Tuple.Create(key, queryStringValue));
```

Comment:
What was expected outcome of the duplicate query parameters `?name=a&name=b`?
Is it the asp.net default: name=a,b which would result in an OR. Or do we pass both `name` tuples through to the Expression parser, which I'm guessing will result in an AND.

Reply:
I believe that I'm following the old behavior and it'll continue being an **AND**. 

I added a test validating that: in HttpRequestExtensionsTests there is a test method named GivenAnUrlWithDuplicatedParameters_WhenEncoded_ThenCompareTheExpectedResults where I'm testing that behavior.

Reply:
Do you believe this is the expected behavior?